staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2331,3 +2331,49 @@ type PreferenceMatcher struct {
 	// +optional
 	Kind string `json:"kind,omitempty"`
 }
+
+// SEVPlatformInfo contains information about the AMD SEV features for the node.
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type SEVPlatformInfo struct {
+	metav1.TypeMeta `json:",inline"`
+	// Base64 encoded platform Diffie-Hellman key.
+	PDH string `json:"pdh,omitempty"`
+	// Base64 encoded SEV certificate chain.
+	CertChain string `json:"certChain,omitempty"`
+}
```

Comment:
I wonder why all of these types have to be defined here as these aren't API object IIUC. Is there a particular reason for that?

Also:
- Why is `metav1.TypeMeta` embedded here?
- Why is `deepCopy` being generated? IIUC it isn't being used anywhere. Further than that, copying such a struct that contains only primitives by-value seems straight forward, so why do we need this?

Reply:
Right, these are not part of the VMI spec API. However they are part of the REST API. According to my observation `staging/src/kubevirt.io/api/core/v1/types.go` already includes structs not used in the VMI spec but used in REST calls (e.g. the Profiler types). Do you think better to move the SEV structs somewhere else?

Regarding `metav1.TypeMeta` and `deepCopy`: this is needed for the code generator:
```
make generate
...
# kubevirt.io/client-go/kubecli
../../vendor/kubevirt.io/client-go/kubecli/vmi.go:482:74: cannot use &sevPlatformInfo (type *"kubevirt.io/api/core/v1".SEVPlatformInfo) as type "k8s.io/apimachinery/pkg/runtime".Object in argument to v.restClient.Get().RequestURI(uri).Do(context.Background()).Into:
	*"kubevirt.io/api/core/v1".SEVPlatformInfo does not implement "k8s.io/apimachinery/pkg/runtime".Object (missing DeepCopyObject method)
make: *** [Makefile:42: generate] Error 1
```

Reply:
> Right, these are not part of the VMI spec API. However they are part of the REST API. According to my observation staging/src/kubevirt.io/api/core/v1/types.go already includes structs not used in the VMI spec but used in REST calls (e.g. the Profiler types). Do you think better to move the SEV structs somewhere else?

Well, not sure about it.
I guess we can keep it there unless @alicefr has another opinion

> Regarding metav1.TypeMeta and deepCopy: this is needed for the code generator:

:ok_hand:

Reply:
mmh it is an interesting point. TBH, I think they fit in this file. One of the reasons why we don't have it in the VMI spec it's because it is a value set a runtime before launching the guest. This reminds me a bit  hotplugged volumes. We report the values in the status but we don't use the type in the VMI spec. Actually, I think it could be reasonable to register the SEVPlatformInfo value in the VMIStatus. @vasiliy-ul wdyt? It can be part of a later PR for me

Reply:
Yep, initially I was thinking about having `SEVPlatformInfo` and `SEVMeasurementInfo` in the `VMIStatus`. This seems reasonable. Just decided to make things simple at first as this could be added in the follow-up PR.

Reply:
sounds reasonable to me. @iholder-redhat ?

Reply:
Sounds great :+1: 
Thanks for the discussion

Reply:
Added this to the tracking issue as a TODO item as a reminder.